app/assets/javascripts/discourse/app/controllers/sidebar-section-form.js

```diff
@@ -329,23 +374,112 @@ export default Controller.extend(ModalFunctionality, {
     return this.model.links.filter((link) => !link._destroy);
   },
 
+  get activeSecondaryLinks() {
+    return this.model.secondaryLinks?.filter((link) => !link._destroy);
+  },
+
   get header() {
     return this.model.id
       ? "sidebar.sections.custom.edit"
       : "sidebar.sections.custom.add";
   },
 
+  @bind
+  reorder(linkFromId, linkTo, above) {
+    if (linkFromId === linkTo.objectId) {
+      return;
+    }
+    let linkFrom = this.model.links.find(
+      (link) => link.objectId === linkFromId
+    );
+    if (!linkFrom) {
+      linkFrom = this.model.secondaryLinks.find(
+        (link) => link.objectId === linkFromId
+      );
+    }
+
+    if (linkFrom.isPrimary) {
+      this.model.links.removeObject(linkFrom);
+    } else {
+      this.model.secondaryLinks?.removeObject(linkFrom);
+    }
+
+    if (linkTo.isPrimary) {
+      const toPosition = this.model.links.indexOf(linkTo);
+      linkFrom.segment = "primary";
+      this.model.links.insertAt(above ? toPosition : toPosition + 1, linkFrom);
+    } else {
+      linkFrom.segment = "secondary";
+      const toPosition = this.model.secondaryLinks.indexOf(linkTo);
+      this.model.secondaryLinks.insertAt(
+        above ? toPosition : toPosition + 1,
+        linkFrom
+      );
+    }
+  },
+
+  get canDelete() {
+    return this.model.id && !this.model.sectionType;
+  },
+
+  @bind
+  deleteLink(link) {
+    if (link.id) {
+      link._destroy = "1";
+    } else {
+      if (link.isPrimary) {
+        this.model.links.removeObject(link);
+      } else {
+        this.model.secondaryLinks.removeObject(link);
+      }
+    }
```

Comment:
```suggestion
    if (link.id) {
      link._destroy = "1";
    } else if (link.isPrimary) {
      this.model.links.removeObject(link);
    } else {
      this.model.secondaryLinks.removeObject(link);
    }
```

Reply:
I think this logic need to stay as nested if. 

If link.id exists, it means that record is in database and we need _destroy param to remove it. It disappears from UI because of that - https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/app/controllers/sidebar-section-form.js#L329

Otherwise, we simply remove the object from primary or secondary collection.